File: past-competitions/2016/src/holly/src/com/lmco/cq2016/Prob00.java
package com.lmco.cq2016;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

public class Prob00 {
    private static final String INPUT_FILE_NAME = "Prob00.in.txt";
    
    public static void main(String[] args) {
        try {
            // prepare to read the file
            InputStream in = Prob00.class.getResourceAsStream(INPUT_FILE_NAME);
            BufferedReader br = new BufferedReader(new InputStreamReader(in));
            
            String inLine = null;
            
            // get the number of test cases
            int T = Integer.parseInt(br.readLine());
            
            // loop through test cases
            while (T-- > 0) {
                // get the number of lines in each test case
                int N = Integer.parseInt(br.readLine());
                
                // loop through the lines
                for (int i=0; i<N; i++) {
                    // read the line of text
                    inLine = br.readLine();
                    
                    // print it out
                    System.out.println(inLine);
                }
            }
            
            // clean up
            br.close();
            in.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: past-competitions/2018/src/brett-cpp/Prob00.c
#include <stdio.h>
#include <stdlib.h>
int main(int argc, char *argv[])
{
   FILE *br;
   char INPUT_FILE_NAME[] = "Prob00.in.txt";
   int T;
   int N;
   int i, j;
   int MAX_STRING = 80;
   char inLine[MAX_STRING+1];

   // prepare to read the file
   br = fopen(INPUT_FILE_NAME, "r");

   // get the number of test cases
   fscanf(br, "%d\n", &T);

   // loop through the test cases
   while (T > 0)
   {
      // get the number of lines in each test case
      fscanf(br, "%d\n", &N);

      // loop through the lines
      for (i=0; i<N; i++)
      {
         // Read the line of text
         j = 0;
         inLine[j] = fgetc(br);
         while ( (j < MAX_STRING) && (inLine[j] != '\n') && (inLine[j] != '\0') && (inLine[j] != '\r') )
         {
            j = j + 1;
            inLine[j] = fgetc(br);
         }
         inLine[j] = '\0';

         // print it out
         printf ("%s\n", inLine);
      }

      T = T - 1;

   }

   // clean up
   fclose(br);

}
